partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldAndTimeBasedPartitioner.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.storage.partitioner;
+
+import io.confluent.connect.storage.common.StorageCommonConfig;
+import io.confluent.connect.storage.errors.PartitionException;
+import io.confluent.connect.storage.util.DataUtils;
+import org.apache.kafka.common.protocol.types.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Schema.Type;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public class FieldAndTimeBasedPartitioner<T> extends TimeBasedPartitioner<T> {
+  private static final Logger log = LoggerFactory.getLogger(FieldAndTimeBasedPartitioner.class);
+  private List<String> fieldNames;
+
+
+  @Override
+  public void configure(Map<String, Object> config) {
+    super.configure(config);
+    fieldNames = (List<String>) config.get(PartitionerConfig.PARTITION_FIELD_NAME_CONFIG);
+    delim = (String) config.get(StorageCommonConfig.DIRECTORY_DELIM_CONFIG);
+  }
+
+  @Override
+  public String encodePartition(SinkRecord sinkRecord) {
+    String timePartition = super.encodePartition(sinkRecord);
+
+    Object value = sinkRecord.value();
+    log.info(value.getClass().toString());
+
+    if (value instanceof Struct) {
```

Comment:
Would `FieldPartitioner` be a better place for the field extraction logic, then we could simply return `fieldPartitioner.encodePartition(sinkRecord) + delim + super.encodePartition(sinkRecord)`